src/js/components/form-elements/v2/Switch.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+
+import Translate from 'utils/Translate';
+
+import './style.scss';
+
+const Switch = ({ defaultValue, titles, onChange }) => {
```

Comment:
this should be ready to receive the `...field` prop from the `react-hook-form`. I can't exactly say now what all the props will be, so for now let's at least add it as `...rest` (let's name it as `...fieldProps`).